sound/soc/sof/pcm.c

```diff
@@ -712,7 +712,9 @@ static int sof_pcm_probe(struct snd_soc_component *component)
 					 SND_SOF_SUSPEND_DELAY);
 	pm_runtime_use_autosuspend(component->dev);
 	pm_runtime_enable(component->dev);
-	err = pm_runtime_idle(component->dev);
+
+	pm_runtime_mark_last_busy(component->dev);
+	err = pm_runtime_put_autosuspend(component->dev);
 	if (err < 0)
```

Comment:
and I am confused here to. I thought we talked about making sure the topology is loaded in D0 but I see no _get?

Reply:
@plbossart we enable runtime_pm for the sof device only in sof_pcm_probe(). So it will be in D0 prior to that anyway. So we dont need a get() or a put_autosuspend() in sof_probe.

Reply:
ok, so that means we never have a race condition as described in https://github.com/thesofproject/linux/issues/193?

Reply:
@plbossart yes, thats correct.